src/toast/cc-toast.js

```diff
@@ -0,0 +1,336 @@
+    this.showProgress = false;
+
+    /** @type {number} The amount of time (in millis) before the `cc-toast:dismiss` event is fired. The timer starts as soon as the component is attached to the DOM. */
+    this.timeout = 5000;
+
+    /** @type {Animation} animation controlling the timout and animating the progress bar. */
+    this._animation = null;
+  }
+
+  /* region PRIVATE METHODS*/
+  _dismiss () {
+    dispatchCustomEvent(this, 'dismiss');
+  }
+
+  _getIcon () {
+    if (this.intent === 'info') {
+      return infoSvg;
+    }
+    if (this.intent === 'success') {
+      return successSvg;
+    }
+    if (this.intent === 'warning') {
+      return warningSvg;
+    }
+    if (this.intent === 'danger') {
+      return dangerSvg;
+    }
+  }
+
+  _getIconAlt () {
+    if (this.intent === 'info') {
+      return i18n('cc-toast.icon-alt.info');
+    }
+    if (this.intent === 'success') {
+      return i18n('cc-toast.icon-alt.success');
+    }
+    if (this.intent === 'warning') {
+      return i18n('cc-toast.icon-alt.warning');
+    }
+    if (this.intent === 'danger') {
+      return i18n('cc-toast.icon-alt.danger');
+    }
+  }
+
+  _onCloseButtonClick () {
+    this._dismiss();
+  }
+
+  _pause () {
+    this._animation?.pause();
+  }
+
+  _resume () {
+    this._animation?.play();
+  }
+  /* endregion*/
+
+  disconnectedCallback () {
+    super.disconnectedCallback();
+    this._animation?.finish();
+    this._animation = null;
+  }
+
+  firstUpdated (_changedProperties) {
+    if (this.timeout <= 0) {
+      return;
+    }
+
+    this._animation = this.shadowRoot.querySelector('.progress-bar-track').animate(
+      [
+        { width: '100%' },
+        { width: 0 },
+      ],
+      {
+        duration: this.timeout,
+        fill: 'forwards',
+        easing: 'linear',
+      },
+    );
+    this._animation.finished.then(() => {
+      this._animation = null;
+      this._dismiss();
+    });
+
+    this._animation.play();
+  }
+
+  render () {
+    const tabIndex = (this.timeout > 0) ? '0' : undefined;
+
+    return html`
+      <div class="toast" 
+           @mouseenter=${this._pause} 
+           @mouseleave=${this._resume} 
+           @focus=${this._pause}
+           @blur=${this._resume}
+           tabindex="${ifDefined(tabIndex)}">
+        <div class="icon-wrapper">
+          <img class="icon" src="${this._getIcon()}" alt="${this._getIconAlt()}">
+        </div>
+        <div class="right">
+          <div class="content">
+            ${this.heading ? html`<div class="heading">${this.heading}</div>` : ''}
+            ${this.message ? html`<div>${this.message}</div>` : ''}
+          </div>
+
+          ${this.closeable ? html`
+            <button class="close-button"
+                    @click=${this._onCloseButtonClick}
+                    @focus=${this._pause}
+                    @blur=${this._resume}
+                    title="${i18n('cc-toast.close')}">
```

Comment:
Let's avoid relying on the `title` attribute alone to generate the accessible name of an element.
Specs say that this should work everywhere but most people say it's a bad idea to rely on it anyway because in reality, its support (as an accessible name) is dodgy.

Rely on the button content (the image in that case) to provide the accessible name:
`<img src="${closeSvg}" alt="${i18n('cc-toast.close')}">`

You can keep the `title` as it provides a tooltip for other users. It is redundant but it's tolerated when dealing with interactive elements that only contain images :wink: 

```js
<button class="close-button"
  @click=${this._onCloseButtonClick}
  @focus=${this._pause}
  @blur=${this._resume}
  title="${i18n('cc-toast.close')}">
    <img src="${closeSvg}" alt="${i18n('cc-toast.close')}">
</button>
```

Reply:
(sorry for not pointing this out earlier, last time I reviewed this part, it was a `cc-button` and it uses an `aria-label` so it was ok)